src/content/blog/ga_qas.md

```diff
@@ -0,0 +1,233 @@
+---
+title: An Introduction to GA-QAS
+author: Vu Tuan Hai
+day: 21
+month: 5
+year: 2024
+tags:
+  - quantum architecture search
+  - python
+  - qiskit
+  - genetic-algorithm
+---
+
+In various quantum optimization problems, choosing the right ansatz is a critical point that will affect the accuracy of the result. Many template ansatzes have been proposed, such as Graph Ansatz and EffecientSU2, … but they are limited in their applicability. Hence, we've created a search engine called **GA-QAS** (Genetic Algorithm for Quantum Architecture Search) to aid in the discovery of the right ansatz. In this post, we will guide you on how to use it efficiently.
+
+Required packages:
+
+- qiskit 1.0+
+- matplotlib
+- tdqm
+
+## Now, let’s start!
+
+# Step 1. Define problem (W state preparation)
+
+<figure>
+    <img src='/public/images/ga-qas/ga-qas1.png' width="600" style="margin: auto" alt='description' />
+    <figcaption>
+    Fig 1. (a) State preparation scheme based on quantum compilation technique, where U, V are ansatz and state that need to prepare[1][2]. (b) Some default ansatz, such as hyper-graph.
+    </figcaption>
+</figure>
+
+In this post, we will use a Python package named <a href="https://github.com/vutuanhai237/qoop">$\langle \mathrm{qo}|\mathrm{op} \rangle$</a>, which is a core package for developing quantum circuit optimization frameworks.
+You can download it via `git` as:
+
+```
+git clone https://github.com/vutuanhai237/qoop.git
+```
+
+Note that you should put it on the same level as your Python/Jupyter Notebook file. The best is:
+
+```
+folder_name
+│   your_python_file.py
+|   your_jupyter_file.ipynb
+└───qoop
+│   └───evolution
+│       │   environment.py
+│       │   cross_over.py
+│       │   ...
+```
+
+First, we define the problem, take example quantum state preparation, more easier W state preparation. 3-qubit W state is defined as:
+
+$$
+|W_3\rangle=V|000\rangle=\frac{1}{\sqrt{3}}(|001\rangle+|010\rangle+|100\rangle)
+$$
+
+And so on for a larger number of qubits. If we want to prepare such a state, we will need to care about the unitary operator $V$, but in some cases, $V$ requires large resources (such as large depth and number of gates). Then, we want to find another unitary $U(\theta)$ that is "equal" to V, $U(\theta)$ requires fewer resources than $V$ but consumes an initial to optimize $\theta$. The problem comes from here: the structure of $U$, or ansatz $U$, affects how "hard" and "large" this optimization process is.
+
+You can try to prepare 3-qubit W state by using $\langle qo|op \rangle$, a wiki can be found [here](https://github.com/vutuanhai237/qoop/wiki/Advance:-Custom-state-preparation):
```

Comment:
Is this comment needed since the code is found immediately below? It's a great idea to link to other documentation, but it's not specific to the state this demo involves. Maybe change this for a lead in/explanation of the code found below.

Reply:
I cleared the duplicate example code, then, edited this paragraph. Now it turns out to be the explanation of the code below.